Libraries/CustomComponents/Navigator/Navigator.js

```diff
@@ -1043,6 +1043,14 @@ var Navigator = React.createClass({
     return this.state.routeStack.slice();
   },
 
+  getCurrentRouteIndex: function() {
+    return this.state.presentedIndex;
+  },
+
+  getCurrentRoute: function() {
+    return {...this.state.routeStack[this.getCurrentRouteIndex()]};
```

Comment:
@hedgerwang I am not familiar with how the navigator will look in future, but for current, I needed this feature for my certain use case. I see some other questionable things about navigator, and I hope they get fixed in future as well.